src/WebUI/src/composables/character/use-inventory-quick-equip.ts

```diff
@@ -0,0 +1,50 @@
+import { type EquippedItemId, type EquippedItemsBySlot } from '@/models/character';
+import { ItemSlot } from '@/models/item';
+import { type UserItem } from '@/models/user';
+import { updateCharacterItems } from '@/services/characters-service';
+import { getAvailableSlotsByItem, weaponSlots } from '@/services/item-service';
+import { useUserStore } from '@/stores/user';
+import { characterKey, characterItemsKey } from '@/symbols/character';
+import { useInventoryEquipment } from '@/composables/character/use-inventory-equipment';
+
+export const useInventoryQuickEquip = (equippedItemsBySlot: Ref<EquippedItemsBySlot>) => {
+  const { user } = toRefs(useUserStore());
+  const character = injectStrict(characterKey);
+  const { loadCharacterItems } = injectStrict(characterItemsKey);
+  const { isEquipItemAllowed, getUnequipItemsLinked } = useInventoryEquipment();
+
+  const onQuickEquip = async (item: UserItem) => {
+    if (!item || !isEquipItemAllowed(item, user.value!.id)) return;
+
+    const availableSlots = getAvailableSlotsByItem(item.item, equippedItemsBySlot.value);
+    const targetSlot = getTargetSlot(availableSlots);
+
+    if (targetSlot) {
+      const items: EquippedItemId[] = [{ userItemId: item.id, slot: targetSlot }];
+
+      await updateItems(items);
+    }
+  };
+
+  const onQuickUnequip = async (slot: ItemSlot) => {
+    const items: EquippedItemId[] = getUnequipItemsLinked(slot, equippedItemsBySlot.value);
+
+    await updateItems(items);
+  };
+
+  const updateItems = async (items: EquippedItemId[]) => {
+    await updateCharacterItems(character.value.id, items);
+    await loadCharacterItems(0, { id: character.value.id });
+  };
+
+  const getTargetSlot = (slots: ItemSlot[]): ItemSlot | undefined => {
+    return slots
+      .filter(slot => weaponSlots.includes(slot) ? !equippedItemsBySlot.value[slot] : true)
```

Comment:
Not critical, but I want you to get the idea )
the service should have this method, for example: `checkIsWeaponbySlot`

just answer your own questions: 
1/ whether the code depends on the framework
2/ is it possible to move this code to the backend at some point (imagine that we can't determine which slot to put an item in, because all the business logic is on the server).